app/models/admin/status_batch_action.rb

```diff
@@ -73,6 +73,7 @@ def handle_mark_as_sensitive!
     # Can't use a transaction here because UpdateStatusService queues
     # Sidekiq jobs
     statuses.includes(:media_attachments, :preview_cards).find_each do |status|
+      next if status.discarded?
```

Comment:
No need to have two different `next` statements, I would just change the `unless` to `if` below.